src/interfaces/challenge.ts

```diff
@@ -0,0 +1,18 @@
+// export enum StatusState {
+//   Progress = 0,
+//   Success,
+//   Failure
+// }
+
+export interface IChallenge {
+  id: number;
+  state: string;
+  startDate: string;
+  tasksOrder: string;
+  tasksStatus: string;
+  achievementsStatus: string;
```

Comment:
Should have a Map<number, string> type. 

https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Map